planetary-variables/soil-water-content/derived-root-zone-soil-water-content/script.js

```diff
@@ -0,0 +1,120 @@
+//VERSION=3
+
+const nDays = 10;           // The number of days to load data for
+const scaleFactor = 1000;   // The scale factor for the SWC values
+const vmin = 0.1;           // The minimum value of the colormap
+const vmax = 0.4;           // The maximum value of the colormap
+
+function setup() {
+    return {
+        input: ["SWC", "dataMask"],
+        output: { bands: 4 },
+        mosaicking: "ORBIT"
+    };
+}
+
+
+function preProcessScenes(collections){
+    collections.scenes.orbits = collections.scenes.orbits.filter(function (orbit) {
+        var orbitDateFrom = new Date(orbit.dateFrom)
+        // Select all images within the last nDays
+        return orbitDateFrom.getTime() >= (collections.to.getTime() - (nDays * 24 * 3600 * 1000));
+    })
+    return collections
+}
+
+
+function expFilter(swc, dataMask, dates, timeConst) {
+    let deltaDays = 0;       // Initialize time difference with previous valid time [Unit days]
+    let gain = 1.0;          // Initialize gain
+    let referenceIndex = -1; // Initialize index of the first day with data
+    let referenceDate = NaN; // Initialize date of the first day with data
+    let referenceSWC = NaN;  // Initialize SWC value of the first day with data
+
+    // Find the first day with data in the time series
+    for (let i = 0; i < dates.length; i++) {
+        if (dataMask[i]) {
+            // Set index, date and SWC initial values to the first date with data
+            referenceIndex = i;
+            referenceDate = dates[i];
+            referenceSWC = swc[i];
+            break;
+        }
+    }
+
+    // Set first day of derived root-zone SWC equal to surface SWC
+    let drzswc = referenceSWC;
+
+    if (referenceIndex > -1) {
+        // Only apply the filter if there is at least one valid value in the time series
+        for (let i = referenceIndex; i < dates.length; i++) {
+            if (dataMask[i]) {
+                deltaDays = (dates[i] - referenceDate) / (24 * 3600 * 1000);
+                gain = gain / (gain + Math.exp(-deltaDays / timeConst));
+                drzswc = drzswc + gain * (swc[i] - drzswc);
+                referenceDate = dates[i]
+            }
+        }
+    }
+    return [drzswc];
+}
+
+
+function updateColormap(vmin, vmax) {
+    const numIntervals = cmap.length;
+    const intervalLength = (vmax - vmin) / (numIntervals - 1);
+    for (let i = 0; i < numIntervals; i++) {
+        cmap[i][0] = vmin + intervalLength * i;
+    }
+}
```

Comment:
Can be removed, due to changes to `ColorRampVisualizer`. Same in the eob.js script
```suggestion
```